gremlin-python/src/main/python/gremlin_python/structure/io/graphsonV4.py

```diff
@@ -637,7 +629,8 @@ def objectify(cls, d, reader):
             properties = reader.to_object(d["properties"])
             if properties is not None:
                 properties = [item for sublist in properties.values() for item in sublist]
-        return Vertex(reader.to_object(d["id"]), d.get("label", "vertex")[0], properties)
+        label = d.get("label", "vertex")
+        return Vertex(reader.to_object(d["id"]), label if isinstance(label, str) else label[0], properties)
```

Comment:
I think this leads to a slight behavior difference with the Java driver as it sets the last item in the list as the label and not the first. This probably doesn't really matter though since its just for backwards compatibility?